Health Alarm to Dynatrace Event implementation (#8476)
netdata/netdata: health/notifications/dynatrace/README.md

<!--
---
title: "Dynatrace"
custom_edit_url: https://github.com/netdata/netdata/edit/master/health/notifications/dynatrace/README.md
---
-->

# Dynatrace

> Dynatrace allows you to receive notification using their Events REST API.

Reviewer: ```suggestion
Dynatrace allows you to receive notifications using their Events REST API.
```

> For more details, see [Events API - POST an event](https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/environment-api/events/post-event/).

Reviewer: ```suggestion
See [the Dynatrace documentation](https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/environment-api/events/post-event/) about POSTing an event in the Events API for more details.
```

You need:

1.  Dynatrace Server. You can use the same on all your Netdata servers but make sure the server is network visible from your Netdata hosts.
The Dynatrace server should be with protocol prefixed (http:// or https://), example https://monitor.illumineit.com

> This is a required parameter.

Reviewer: ```suggestion
1.  Dynatrace Server. You can use the same on all your Netdata servers but make sure the server is network visible from your Netdata hosts.
The Dynatrace server should be with protocol prefixed (`http://` or `https://`). For example: `https://monitor.example.com`
This is a required parameter.
```

2.  API Token. Generate a secure access API token that enables access to your Dynatrace monitoring data via the REST-based API.
Generate a Dynatrace API authentication token. On your Dynatrace server goto Settings --> Integration --> Dynatrace API --> Generate token
For more details, see [Dynatrace API - Authentication](https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/basics/dynatrace-api-authentication/).

> This is a required parameter.

Reviewer: ```suggestion
2.  API Token. Generate a secure access API token that enables access to your Dynatrace monitoring data via the REST-based API.
Generate a Dynatrace API authentication token. On your Dynatrace server, go to **Settings** --> **Integration** --> **Dynatrace API** --> **Generate token**.
See [Dynatrace API - Authentication](https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/basics/dynatrace-api-authentication/) for more details.
This is a required parameter.
```

3.  API Space. This is the URL part of the page you have access in order to generate the API Token. For example, for my generated API Token the URL is:
https://monitor.illumineit.com/e/2a93fe0e-4cd5-469a-9d0d-1a064235cfce/#settings/integration/apikeys;gf=all
In that case, my space is _2a93fe0e-4cd5-469a-9d0d-1a064235cfce_
This is a required parameter.

> 4. Generate a Server Tag. On your Dynatrace Server go to Settings --> Tags --> Manually applied tags and create the Tag

Reviewer: ```suggestion
4. Generate a Server Tag. On your Dynatrace Server, go to **Settings** --> **Tags** --> **Manually applied tags** and create the Tag.
```

The Netdata alarm is sent as a Dynatrace Event to be correlated with all those hosts tagged with this Tag you have created.
This is a required parameter.
5. Specify the Dynatrace event. This can be one of CUSTOM_INFO, CUSTOM_ANNOTATION, CUSTOM_CONFIGURATION, CUSTOM_DEPLOYMENT
The default value is CUSTOM_INFO

> This is a required parameter.

Reviewer: ```suggestion
5. Specify the Dynatrace event. This can be one of `CUSTOM_INFO`, `CUSTOM_ANNOTATION`, `CUSTOM_CONFIGURATION`, and `CUSTOM_DEPLOYMENT`. 
The default value is `CUSTOM_INFO`.
This is a required parameter.
```

6. Specify the annotation type. Practically this is the source of the Dynatrace event. Put whatever it fits you, for example 

> _NetData Alarm_ that is also the default value.

Reviewer: ```suggestion
6. Specify the annotation type. This is the source of the Dynatrace event. Put whatever it fits you, for example, 
_Netdata Alarm_, which is also the default value.
```

<!--
---
title: "Dynatrace"
custom_edit_url: https://github.com/netdata/netdata/edit/master/health/notifications/dynatrace/README.md
---
-->

# Dynatrace

Dynatrace allows you to receive notification using their Events REST API.

> See here for more details:

Reviewer 2: ```suggestion
For more details, see [Events API - POST an event](https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/environment-api/events/post-event/).
```

> https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/environment-api/events/post-event/

Reviewer 2: ```suggestion

```

You need:

> 1.  The Dynatrace Server. You can use the same on all your Netdata servers but make sure the server is network visible from your NetData hosts.

Reviewer 2: ```suggestion
1.  Dynatrace Server. You can use the same on all your Netdata servers but make sure the server is network visible from your Netdata hosts.
```

The Dynatrace server should be with protocol prefixed (http:// or https://), example https://monitor.illumineit.com
This is a required parameter.
2.  API Token. Generate a secure access API token that enables access to your Dynatrace monitoring data via the REST-based API.
Generate a Dynatrace API authentication token. On your Dynatrace server goto Settings --> Integration --> Dynatrace API --> Generate token

> For more on this, read https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/basics/dynatrace-api-authentication/ 

Reviewer 2: ```suggestion
For more details, see [Dynatrace API - Authentication](https://www.dynatrace.com/support/help/extend-dynatrace/dynatrace-api/basics/dynatrace-api-authentication/).
```

This is a required parameter.

> 3.  The API Space. This is the URL part of the page you have access in order to generate the API Token. For example, for my generated API Token the URL is:

Reviewer 2: ```suggestion
3.  API Space. This is the URL part of the page you have access in order to generate the API Token. For example, for my generated API Token the URL is:
```

https://monitor.illumineit.com/e/2a93fe0e-4cd5-469a-9d0d-1a064235cfce/#settings/integration/apikeys;gf=all
In that case, my space is _2a93fe0e-4cd5-469a-9d0d-1a064235cfce_
This is a required parameter.
4. Generate a Server Tag. On your Dynatrace Server go to Settings --> Tags --> Manually applied tags and create the Tag

> The NetData alarm will be sent as a Dynatrace Event to be correlated with all those hosts tagged with this Tag you have created.

Reviewer 2: ```suggestion
The Netdata alarm is sent as a Dynatrace Event to be correlated with all those hosts tagged with this Tag you have created.
```
